uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/apis/leave-request-api_test.js

```diff
@@ -21,6 +25,34 @@ define([
       //Intercept backend calls for LeaveRequest.balanceChangeByAbsenceType
       $httpBackend.whenGET(/action\=getbalancechangebyabsencetype&entity\=LeaveRequest/)
         .respond(mockData.balanceChangeByAbsenceType());
+
+      //Intercept backend calls for LeaveRequest.calculateBalanceChange
+      $httpBackend.whenGET(/action\=calculatebalancechange&entity\=LeaveRequest/)
+        .respond(mockData.calculateBalanceChange());
+
+      //Intercept backend calls for LeaveRequest.create in POST
+      $httpBackend.whenPOST(/\/civicrm\/ajax\/rest/)
+        .respond(function (method, url, data, headers, params) {
+          var uriParts = decodeURIComponent(data).split('&');
```

Comment:
There's a lot of stuff going here, why's that? In any case, please put all this logic in a separate function